packages/data-context/src/sources/MigrationDataSource.ts

```diff
@@ -118,7 +138,13 @@ export class MigrationDataSource {
     return this.componentTestingMigrationStatus
   }
 
-  async supportFilesForMigrationGuide (): Promise<FilesForMigrationUI> {
+  async supportFilesForMigrationGuide (): Promise<FilesForMigrationUI | null> {
+    const config = await this.parseCypressConfig()
+
+    if (!shouldShowRenameSupport(config)) {
+      return null
+    }
+
     if (!this.ctx.currentProject) {
       throw Error(`Need this.ctx.projectRoot!`)
```

Comment:
How is this error handled?

Reply:
We have a global `catch` in the data context (somewhere, forgot where), which propogates the errors to the front-end in the GraphQL request that caused them.

We have this check (`if (!this.ctx.currentProject)` in about 50 places, it's really not ideal. I think we probably should have a getter that handles this, like

```ts
get currentProject () {
  if (!this.ctx.currentProject) {
    throw Error('Current Project should be initialized prior to this point in the application')
  }
  return this.ctx.currentProject
}
```

Reply:
We have this check in at least 19 places, I made a ticket to reduce them to a single location: https://cypress-io.atlassian.net/browse/UNIFY-988